templates/manila/bin/healthcheck.py

```diff
@@ -0,0 +1,242 @@
+        cls.binary = 'manila-' + binary
+        cls.config_dir = config_dir
+
+        if binary != 'share':
+            services_filters = [
+                {'host': CONF.host,
+                 'state': STATE_UP,
+                 'status': STATUS_ENABLED,
+                 'binary': cls.binary}]
+        else:
+            services_filters = []
+            for backend in CONF.enabled_share_backends:
+                host = "%s@%s" % (CONF.host, backend)
+                services_filters.append({'host': host,
+                                         'status': STATUS_ENABLED,
+                                         'binary': cls.binary})
+
+        # Store the filters used to find out services.
+        cls.services_filters = services_filters
+
+    @staticmethod
+    def check_service(services, **filters):
+        # Check if services were already created and are registered in the db
+        our_services = [service for service in services
+                        if all(service.get(key, None) == value
+                               for key, value in filters.items())]
+        if not our_services:
+            raise ServiceNotFoundException(filters=filters)
+
+        num_services = len(our_services)
+        if num_services != 1:
+            print(f'{num_services} services found!: {our_services}')
+
+        ok = all(service['state'] == STATE_UP for service in our_services)
+        if not ok:
+            raise ServiceNotUp()
+
+    def _parse_service_list_output(self, services_list):
+        logging.exception(f'DEBUG: services list output: {services_list}')
+        # Decode the characters, remove the line splits and the
+        # blank spaces
+        services_list = services_list.decode('utf-8').split('\n')
+        parsed_service_list = []
+        for service in services_list:
+            if "DEBUG" in service:
+                continue
+            parsed_service_list.append(service)
+
+        if not parsed_service_list:
+            raise ServiceNotFoundException(filters={})
+
+        parsed_service_list = [
+            service.split() for service in parsed_service_list]
+
+        # First line is related to the headers. We don't care about the
+        # updated at, so we drop it and sanitize the strings.
+        logging.exception(
+            f'DEBUG: parsed services list: {parsed_service_list}')
+        services_list_headers = parsed_service_list[0]
+        columns_to_ignore_starts_at = (
+            services_list_headers.index("Updated")
+            if "Updated" in services_list_headers else None)
+
+        if not columns_to_ignore_starts_at:
+            raise ServiceNotFoundException(filters={})
+
+        services_list_headers = (
+            services_list_headers[0:columns_to_ignore_starts_at])
+        services_list_headers = (
+            [header.lower() for header in services_list_headers])
+        logging.exception(
+            f'DEBUG: service list headers: {services_list_headers}')
+
```

Comment:
can we drop all these exception logs now? 

iirc we had some trouble getting logs to show up that made us try this route?

Reply:
I added this only to help debugging in the CI, as I was not being able to reproduce this issue in my environment. It can be dropped now :)